source/extensions/common/aws/signer_impl.cc

```diff
@@ -24,8 +24,14 @@ void SignerImpl::sign(Http::RequestMessage& message, bool sign_body) {
 }
 
 void SignerImpl::sign(Http::RequestHeaderMap& headers) {
-  // S3 payloads require special treatment.
-  if (service_name_ == "s3") {
+  // S3, Glacier, ES payloads require special treatment.
+  // S3:
+  // https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html.
+  // ES:
+  // https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-request-signing.html.
+  // Glacier:
+  // https://docs.aws.amazon.com/amazonglacier/latest/dev/amazon-glacier-signing-requests.html.
+  if (service_name_ == "s3" || service_name_ == "glacier" || service_name_ == "es") {
```

Comment:
Sorry, I think we can have this decision at initialization. Let's have a `const bool` member variable (not sure about the name, e.g. `require_special_treatment_`). We can initialize that bool with `require_special_treatment_(service_name_ == "s3" || service_name_ == "glacier" || service_name_ == "es")` and use that here.

Reply:
+1

Reply:
ok